x-pack/plugins/observability/public/embeddable/slo/slo_embeddable.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Subscription } from 'rxjs';
+
+import {
+  Embeddable as AbstractEmbeddable,
+  EmbeddableOutput,
+  IContainer,
+} from '@kbn/embeddable-plugin/public';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import dateMath from '@kbn/datemath';
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { SloOverview } from './slo_overview';
+import type { SloEmbeddableDeps, SloEmbeddableInput } from './types';
+
+export const SLO_EMBEDDABLE = 'SLO_EMBEDDABLE';
+
+function datemathToEpochMillis(
```

Comment:
I took this from [infra](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/utils/datemath.ts). It didn't make sense to add infra as a required plugin and import from there